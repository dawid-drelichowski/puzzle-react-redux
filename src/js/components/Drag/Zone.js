import React, {PropTypes} from 'react';
import Item from 'components/Drag/Item';

export default class extends React.Component {
  static propTypes = {
    piecesOrder: PropTypes.arrayOf(PropTypes.number).isRequired,
    onDragStart: PropTypes.func.isRequired,
    onDragEnd: PropTypes.func.isRequired
  };
  onDragStart = (event) => {
    const target = event.target,
      dataTransfer = event.dataTransfer;

    dataTransfer.effectAllowed = 'move';
    dataTransfer.dropEffect = 'move';
    dataTransfer.setData('text', '1'); //Firefox effectAllowed and dropEffect bug fix

    target.classList.add('draggable-current');
    this.props.onDragStart(this.getFieldFromElement(target));
  };
  onDragEnd = (event) => {
    const target = event.target;

    target.classList.remove('draggable-current');
    this.props.onDragEnd(this.getFieldFromElement(target));
  };
  render() {
    return <div className="col-lg-5 label label-danger">
      <ul
        className="grid label-danger"
        onDragStart={this.onDragStart}
        onDragEnd={this.onDragEnd}
      >
        {this.props.piecesOrder.map((value, index) => {
          return <Item key={index} field={value} />;
        })}
      </ul>
    </div>;
  }
  getFieldFromElement(element) {
    return parseInt(element.getAttribute('data-field'), 10);
  }
}
