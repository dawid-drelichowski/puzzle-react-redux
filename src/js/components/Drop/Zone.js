import React, {PropTypes} from 'react';
import Item from 'containers/Drop/Item';

export default class extends React.Component {
  static propTypes = {
    piecesCount: PropTypes.number.isRequired,
  };
  render() {
    const pieces = new Array(this.props.piecesCount).fill(true);

    return <div className="col-lg-5 label label-success">
      <ul className="grid label-success">
        {pieces.map((value, index) => {
          return <Item key={index} field={index + 1} />
        })}
      </ul>
    </div>;
  }
}