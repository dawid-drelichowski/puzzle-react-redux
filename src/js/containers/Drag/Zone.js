import {connect} from 'react-redux';
import Creators from 'actions/index';
import Zone from 'components/Drag/Zone';

export default connect(
  () => {
    return {};
  },
  dispatch => {
    return {
      onDragStart: index => {
        dispatch(Creators.dragStart(index));
      },
      onDragEnd: index => {
        dispatch(Creators.dragEnd(index));
      }
    };
  }
)(Zone);
