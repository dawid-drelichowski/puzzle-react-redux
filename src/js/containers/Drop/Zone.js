import {connect} from 'react-redux';
import Creators from 'actions/index';
import Zone from 'components/Drop/Zone';

export default connect(
  state => {
    return {currentlyDragged: state.currentlyDragged};
  },
  dispatch => {
    return {
      onDrop: index => {
        dispatch(Creators.dropped(index));
      }
    };
  }
)(Zone);
