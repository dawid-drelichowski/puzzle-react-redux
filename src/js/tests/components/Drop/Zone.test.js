import React from 'react';
import Zone from 'components/Drop/Zone';
import renderer from 'react-test-renderer';

describe('Drop zone component', () => {
  function getComponent(piecesCount, currentlyDragged = 0, onDrop = () => {}) {
    return renderer.create(
      <Zone piecesCount={piecesCount} currentlyDragged={currentlyDragged} onDrop={onDrop} />
    );
  }

  it('should has 2 items', () => {
    expect(getComponent(2).toJSON()).toMatchSnapshot();
  });

  it('should has 4 items', () => {
    expect(getComponent(4).toJSON()).toMatchSnapshot();
  });
});