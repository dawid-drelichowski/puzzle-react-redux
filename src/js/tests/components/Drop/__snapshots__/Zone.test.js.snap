// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop zone component should have 2 items 1`] = `
<div
  className="col-lg-5 label label-success"
>
  <ul
    className="grid label-success"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <li
      data-valid={1}
    />
    <li
      data-valid={2}
    />
  </ul>
</div>
`;

exports[`Drop zone component should have 4 items 1`] = `
<div
  className="col-lg-5 label label-success"
>
  <ul
    className="grid label-success"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <li
      data-valid={1}
    />
    <li
      data-valid={2}
    />
    <li
      data-valid={3}
    />
    <li
      data-valid={4}
    />
  </ul>
</div>
`;
